generative/maisi/scripts/train_controlnet.py

```diff
@@ -0,0 +1,274 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import logging
+import os
+import sys
+import time
+from datetime import timedelta
+from pathlib import Path
+
+import torch
+import torch.distributed as dist
+import torch.nn.functional as F
+from monai.networks.utils import copy_model_state
+from monai.utils import RankFilter
+from torch.cuda.amp import GradScaler, autocast
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.utils.tensorboard import SummaryWriter
+from utils import binarize_labels, define_instance, prepare_maisi_controlnet_json_dataloader, setup_ddp
+
+
+def main():
+    parser = argparse.ArgumentParser(description="maisi.controlnet.training")
+    parser.add_argument(
+        "-e",
+        "--environment-file",
+        default="./config/environment_maisi_controlnet_train.json",
+        help="environment json file that stores environment path",
+    )
+    parser.add_argument(
+        "-c",
+        "--config-file",
+        default="./config/config_maisi_controlnet_train.json",
+        help="config json file that stores hyper-parameters",
+    )
+    parser.add_argument("-g", "--gpus", default=1, type=int, help="number of gpus per node")
+    args = parser.parse_args()
+
+    # Step 0: configuration
+    logger = logging.getLogger("maisi.controlnet.training")
+    # whether to use distributed data parallel
+    use_ddp = args.gpus > 1
+    if use_ddp:
+        rank = int(os.environ["LOCAL_RANK"])
+        world_size = int(os.environ["WORLD_SIZE"])
+        device = setup_ddp(rank, world_size)
+        logger.addFilter(RankFilter())
+    else:
+        rank = 0
+        world_size = 1
+        device = torch.device(f"cuda:{rank}")
+
+    torch.cuda.set_device(device)
+    logger.info(f"Number of GPUs: {torch.cuda.device_count()}")
+    logger.info(f"World_size: {world_size}")
+
+    env_dict = json.load(open(args.environment_file, "r"))
+    config_dict = json.load(open(args.config_file, "r"))
+
+    for k, v in env_dict.items():
+        setattr(args, k, v)
+    for k, v in config_dict.items():
+        setattr(args, k, v)
+
+    # initialize tensorboard writer
+    if rank == 0:
+        tensorboard_path = os.path.join(args.tfevent_path, args.exp_name)
+        Path(tensorboard_path).mkdir(parents=True, exist_ok=True)
+        tensorboard_writer = SummaryWriter(tensorboard_path)
+
+    # Step 1: set data loader
+    train_loader, _ = prepare_maisi_controlnet_json_dataloader(
+        json_data_list=args.json_data_list,
```

Comment:
I didn't find where define this json_data_list, did you run the training script locally?

Reply:
Yeah, I tested it locally. `json_data_list` is defined in the environment config. https://github.com/Project-MONAI/tutorials/blob/3234841a5b44562359cdb3ecb2f00d7cc7564e42/generative/maisi/configs/environment_maisi_controlnet_train.json#L10

Reply:
Yes, I mean that maybe we need some instruction here to tell users how to download the data?
Such as check if the json file and data dir is not exist, download them using `download_and_extract`.
https://github.com/Project-MONAI/MONAI/blob/3a0c2d5afcc91379819fd8428fa83b00fb18c9fd/monai/apps/utils.py#L304

What do you think?

Reply:
Yeah, we do need instructions. I will add instructions about the controlnet training and data in another PR (https://github.com/Project-MONAI/tutorials/pull/1743).